src/libDirectoryService/ViewChangePostProcessing.cpp

```diff
@@ -0,0 +1,413 @@
+            }
+
+            P2PComm::GetInstance().SendBroadcastMessage(shard_peers,
+                                                        vcblock_message);
+            p++;
+        }
+    }
+}
+
+void DirectoryService::ProcessViewChangeConsensusWhenDone()
+{
+    LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+              "View change consensus is DONE!!!");
+
+    m_pendingVCBlock->SetCoSignatures(*m_consensusObject);
+
+    unsigned int index = 0;
+    unsigned int count = 0;
+
+    vector<PubKey> keys;
+    for (auto& kv : m_mediator.m_DSCommitteePubKeys)
+    {
+        if (m_pendingVCBlock->GetB2().at(index) == true)
+        {
+            keys.push_back(kv);
+            count++;
+        }
+        index++;
+    }
+
+    // Verify cosig against vcblock
+    shared_ptr<PubKey> aggregatedKey = MultiSig::AggregatePubKeys(keys);
+    if (aggregatedKey == nullptr)
+    {
+        LOG_GENERAL(WARNING, "Aggregated key generation failed");
+    }
+
+    vector<unsigned char> message;
+    m_pendingVCBlock->GetHeader().Serialize(message, 0);
+    m_pendingVCBlock->GetCS1().Serialize(message, VCBlockHeader::SIZE);
+    BitVector::SetBitVector(message, VCBlockHeader::SIZE + BLOCK_SIG_SIZE,
+                            m_pendingVCBlock->GetB1());
+    if (Schnorr::GetInstance().Verify(message, 0, message.size(),
+                                      m_pendingVCBlock->GetCS2(),
+                                      *aggregatedKey)
+        == false)
+    {
+        LOG_GENERAL(WARNING, "cosig verification fail");
+        return;
+    }
+
+    Peer newLeaderNetworkInfo;
+    unsigned char viewChangeState;
+    {
+        lock_guard<mutex> g(m_mutexPendingVCBlock);
+
+        newLeaderNetworkInfo
+            = m_pendingVCBlock->GetHeader().GetCandidateLeaderNetworkInfo();
+        viewChangeState = m_pendingVCBlock->GetHeader().GetViewChangeState();
+    }
+
+    // StoreVCBlockToStorage(); TODO
+    unsigned int offsetToCandidateLeader = 1;
+
+    Peer expectedLeader;
+    if (m_mediator.m_DSCommitteeNetworkInfo.at(offsetToCandidateLeader)
+        == Peer())
+    {
+        // I am 0.0.0.0
+        expectedLeader = m_mediator.m_selfPeer;
+    }
+    else
+    {
+        expectedLeader
+            = m_mediator.m_DSCommitteeNetworkInfo.at(offsetToCandidateLeader);
+    }
+
+    if (expectedLeader == newLeaderNetworkInfo)
+    {
+        if (newLeaderNetworkInfo == m_mediator.m_selfPeer)
+        {
+            LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+                      "After view change, I am the new leader!");
+            m_mode = PRIMARY_DS;
+        }
+        else
+        {
+            LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+                      "After view change, I am ds backup");
+            m_mode = BACKUP_DS;
+        }
+
+        // Kick ousted leader to the back of the queue, waiting to be eject at
```

Comment:
Discuss with @ckyang , once he merges pow1/pow2.

Reply:
Will be discussed in his PR once it is ready